scala/scala.bzl

```diff
@@ -128,6 +130,20 @@ def _compile(ctx, _jars, dep_srcjars, buildijar):
       out=ctx.outputs.jar.path
       )
   ctx.file_action(output = scalac_args_file, content = scalac_args)
+  javac_sources_cmd = ""
+  compile_java_srcs = len(java_srcs) != 0
+  if (compile_java_srcs):
+  # Set up the args to pass to javac because they can be too long for bash
+    javac_args_file = ctx.new_file(ctx.outputs.jar, ctx.label.name + "_javac_args")
+    javac_args = """ -classpath "{jars}:{out}_tmp" -d {out}_tmp {files}""".format(
+      jars=":".join([j.path for j in jars]),
+      files=" ".join([f.path for f in java_srcs]),
+      out=ctx.outputs.jar.path
+      )
+    ctx.file_action(output = javac_args_file, content = javac_args)
+    javac_sources_cmd = """
+    cat {javac_args} {{out}}_args/files_from_jar > {{out}}_args/java_args
+    env JAVACMD={{java}} {javac} {{jvm_flags}} @{{out}}_args/java_args""".format(javac_args = javac_args_file.path,javac=ctx.file._javac.path)
```

Comment:
What do you mean by broken value?
On יום ב׳, 20 ביוני 2016 at 6:45 P. Oscar Boykin <notifications@github.com>
wrote:

> In scala/scala.bzl
> <https://github.com/bazelbuild/rules_scala/pull/71#discussion_r67633532>:
>
> > @@ -128,6 +130,20 @@ def _compile(ctx, _jars, dep_srcjars, buildijar):
> >        out=ctx.outputs.jar.path
> >        )
> >    ctx.file_action(output = scalac_args_file, content = scalac_args)
> > +  javac_sources_cmd = ""
> > +  compile_java_srcs = len(java_srcs) != 0
> > +  if (compile_java_srcs):
> > +  # Set up the args to pass to javac because they can be too long for bash
> > +    javac_args_file = ctx.new_file(ctx.outputs.jar, ctx.label.name + "_javac_args")
> > +    javac_args = """ -classpath "{jars}:{out}_tmp" -d {out}_tmp {files}""".format(
> > +      jars=":".join([j.path for j in jars]),
> > +      files=" ".join([f.path for f in java_srcs]),
> > +      out=ctx.outputs.jar.path
> > +      )
> > +    ctx.file_action(output = javac_args_file, content = javac_args)
> > +    javac_sources_cmd = """
> > +    cat {javac_args} {{out}}_args/files_from_jar > {{out}}_args/java_args
> > +    env JAVACMD={{java}} {javac} {{jvm_flags}} @{{out}}_args/java_args""".format(javac_args = javac_args_file.path,javac=ctx.file._javac.path)
>
> this environmental variable is used by scalac (JAVACMD) is the same used
> by javac? Did you test with a broken value to see if that fails?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/rules_scala/pull/71/files/99bbd657a1f504b46d6f246149d8321de56abf24#r67633532>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABUIF5yj5LwgjF2aorsn3yZBpQuMsd8lks5qNgzzgaJpZM4Ir-mk>
> .
>